tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in0_sender_receiver_padding_block_sharded.cpp

```diff
@@ -78,11 +86,38 @@ void kernel_main() {
     noc_semaphore_set(in0_mcast_receiver_semaphore_addr_ptr, VALID);
 
     cb_reserve_back(cb_id_in2, batch * in0_block_num_tiles);
-    uint32_t local_read_addr = get_read_ptr(cb_id_in2);
+
+    uint32_t local_read_addr;
+    if constexpr (in4_block_num_tiles > 0) {
+        cb_reserve_back(cb_id_in4, batch * in4_block_num_tiles);
```

Comment:
The cb we mcast to should be receiving `in0_block_num_tiles`

Reply:
right